lib/inferno/dsl/fhir_resource_validation.rb

```diff
@@ -175,10 +175,25 @@ def resource_is_valid?(resource, profile_url, runnable, add_messages_to_runnable
           begin
             response = call_validator(resource, profile_url)
           rescue StandardError => e
-            # This could be a complete failure to connect (validator isn't running)
-            # or a timeout (validator took too long to respond).
             runnable.add_message('error', e.message)
-            raise Inferno::Exceptions::ErrorInValidatorException, "Unable to connect to validator at #{url}."
+
+            # add a specific error message for specific network problems to help in troubleshooting
+            error_message = case e
+                            when Faraday::ConnectionFailed
+                              "Connection failed to validator at #{url}."
+                            when Faraday::TimeoutError
+                              "Timeout while connecting to validator at #{url}."
+                            when Faraday::SSLError
+                              "SSL error connecting to validator at #{url}."
+                            when Faraday::ClientError  # these are 400s
+                              "Client error (4xx) connecting to validator at #{url}."
+                            when Faraday::ServerError  # these are 500s
+                              "Server error (5xx) from validator at #{url}."
+                            else
+                              "Unable to connect to validator at #{url}."
```

Comment:
Sorry one more thing. As I recall the issue that started all of this was actually someone was working on a test kit and passed in a FHIR::Resource instead of a dict into the validate assertion, leading to a no method found error, or whatever it's called in Ruby. I don't want to specifically catch that kind of error here, but if we hit this "else" case I think something specific about the error should be included in the message for easier debugging

Reply:
Ok.  I put a log statement in both places.

I also moved the resource_validator into a method because I got a cyclometric complexity fail on rubocop on the evaluator and they should both maybe match style.